azure-kusto-ingest/azure/kusto/ingest/resource_manager.py

```diff
@@ -0,0 +1,99 @@
+"""
+    This module is serve as a cache to all resources needed by the kusto ingest client
+"""
+
+from azure.kusto.data import KustoClient
+from .connection_string import _ConnectionString
+
+from datetime import datetime
+from datetime import timedelta
+import random
+
+class IngestClientResources:
+    def __init__(self,
+                 secured_ready_for_aggregation_queues = None,
+                 failed_ingestions_queues = None,
+                 successful_ingestions_queues = None,
+                 containers = None,
+                 status_table = None
+                 ):
+        self.secured_ready_for_aggregation_queues = secured_ready_for_aggregation_queues
+        self.failed_ingestions_queues = failed_ingestions_queues
+        self.successful_ingestions_queues = successful_ingestions_queues
+        self.containers = containers
+        self.status_table = status_table
+    
+    def is_applicable(self):
+        if self.secured_ready_for_aggregation_queues \
+           and self.failed_ingestions_queues \
+           and self.successful_ingestions_queues \
+           and self.containers \
+           and self.status_table:
+            return True
+        return False
+
+class _ResourceManager:
+    def __init__(self, 
+                 kusto_client):
+        self._kusto_client = kusto_client
+        self._cache_lifetime = timedelta(hours = 5)
```

Comment:
Let's refresh every hour, like it is in C#.

Reply:
Drop the spaces in `hours = 5`.

Reply:
It seems to me that in C# it is also 5 hours, check the param 'c_cacheLifetime' in the ResourceManager.cs

Reply:
My mistake, confused with the wrong argument.
Should have looked at 'c_refreshPeriod', that it is 1 hour.
Changed also the name for compatibility.